src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -96,12 +96,31 @@ internal interface IMessageSource
     internal abstract class HostEnvironmentBase<TEnv> : ChannelProviderBase, IHostEnvironment, IChannelProvider
         where TEnv : HostEnvironmentBase<TEnv>
     {
+        public bool IsCanceled { get; protected set; }
```

Comment:
You are splitting the properties and fields of this class up. They were all declared after the nested types before your change - right above the constructor. Now you have some up here, and most down there. Please keep them grouped together.